### ISSUE TITLE
BIG-IP clustering DeviceGroup owner does not support IPv6

### ISSUE THREAD

**Issue by cbascom** [F5Networks/f5-declarative-onboarding#198]
### Environment
 * Declarative Onboarding Version: 1.18.0-4
 * BIG-IP Version: 15.1.0.5 0.0.8

### Summary
BIG-IP clustering does not work with IPv6 addresses. The DeviceGroup class requires hostname format for its owner value which means an IPv6 address here causes a bad declaration error.

I am able to work around this issue by changing the DeviceGroup owner property to use a JSON pointer: /Common/trust/remoteHost.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
   "async":true,
   "schemaVersion":"1.18.0",
   "Common":{
      "trust":{
         "localUsername":"admin",
         "remoteHost":"fd32:f5:0:a70::2aa1",
         "localPassword":"admin",
         "remotePassword":"admin",
         "remoteUsername":"admin",
         "class":"DeviceTrust"
      },
      "f5test-ha-dg":{
         "autoSync":true,
         "fullLoadOnSync":false,
         "saveOnAutoSync":false,
         "members":[
            "fd32:f5:0:a70::2aa1",
            "fd32:f5:0:a70::2aa2"
         ],
         "asmSync":false,
         "owner":"fd32:f5:0:a70::2aa1",
         "networkFailover":true,
         "type":"sync-failover",
         "class":"DeviceGroup"
      },
      "failoverAddress":{
         "class":"FailoverUnicast",
         "address":"fd32:f5:0:a70::2aa1"
      },
      "class":"Tenant",
      "configsync":{
         "configsyncIp":"fd32:f5:0:a70::2aa1",
         "class":"ConfigSync"
      }
   },
   "class":"Device",
   "label":"BVT BIG-IP declaration for DO"
}
```

2. Observe the following error response:
```json
{
    "code": 400,
    "message": "request failed with null exception",
    "originalRequestBody":"{\"id\":\"2294a082-f98a-47d8-ba86-c0314514bf21\",\"selfLink\":\"https://localhost/mgmt/shared/declarative-onboarding/task/2294a082-f98a-47d8-ba86-c0314514bf21\",\"code\":400,\"status\":\"ERROR\",\"message\":\"bad declaration\",\"errors\":[{\"keyword\":\"format\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf/0/format\",\"params\":{\"format\":\"hostname\"},\"message\":\"should match format \\\"hostname\\\"\"},{\"keyword\":\"format\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf/1/format\",\"params\":{\"format\":\"json-pointer\"},\"message\":\"should match format \\\"json-pointer\\\"\"},{\"keyword\":\"anyOf\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf\",\"params\":{},\"message\":\"should match some schema in anyOf\"}],\"result\":{\"class\":\"Result\",\"code\":400,\"status\":\"ERROR\",\"message\":\"bad declaration\",\"errors\":[{\"keyword\":\"format\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf/0/format\",\"params\":{\"format\":\"hostname\"},\"message\":\"should match format \\\"hostname\\\"\"},{\"keyword\":\"format\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf/1/format\",\"params\":{\"format\":\"json-pointer\"},\"message\":\"should match format \\\"json-pointer\\\"\"},{\"keyword\":\"anyOf\",\"dataPath\":\".declaration.Common['f5test-ha-dg'].owner\",\"schemaPath\":\"dsc.schema.json#/allOf/3/then/properties/owner/anyOf\",\"params\":{},\"message\":\"should match some schema in anyOf\"}]},\"declaration\":{\"async\":true,\"schemaVersion\":\"1.18.0\",\"Common\":{\"trust\":{\"localUsername\":\"admin\",\"remoteHost\":\"fd32:f5:0:
}
```

### Expected Behavior
The declaration should be accepted as valid and the BIG-IP cluster created just as it is when using IPv4 addresses.

### Actual Behavior
The DO schema causes a bad declaration error by enforcing hostname format on the DeviceGroup owner property.

In addition, the error response returned in this case needs improvement. The originalRequestBody parameter in the error response is unusable for debugging anything since it is truncated.

**Comment by dstokesf5** [comment]
I have added this to our internal product backlog as AUTOTOOL-2215. Thank you for reporting it.

**Comment by dstokesf5** [comment]
This issue was resolved in the 1.20.0 release.
